src/fluree/db/flake/flake_db.cljc

```diff
@@ -292,6 +293,131 @@
           (merge-flakes t-new all-flakes)
           (assoc :commit commit-metadata)))))
 
+(defn- component->user-value
+  "Convert an internal pattern component to user-readable format"
+  [db component context]
+  (cond
+    (where/unmatched-var? component)
+    (str (where/get-variable component))
+
+    (where/matched-iri? component)
+    (let [iri (where/get-iri component)]
+      ;; Try to find matching prefix in context
+      (if-let [[prefix ns-iri] (some (fn [[k v]]
+                                       (when (and (string? v) (str/starts-with? iri v))
+                                         [k v]))
+                                     context)]
+        (str prefix ":" (subs iri (count ns-iri)))
+        iri))
+
+    (where/matched-value? component)
+    (where/get-value component)
+
+    (where/matched-sid? component)
+    (let [sid (where/get-sid component db)
+          iri (iri/decode-sid db sid)]
+      (if-let [[prefix ns-iri] (some (fn [[k v]]
+                                       (when (and (string? v) (str/starts-with? iri v))
+                                         [k v]))
+                                     context)]
+        (str prefix ":" (subs iri (count ns-iri)))
+        iri))
+
+    :else
+    (str component)))
+
+(defn- pattern->user-format
+  "Convert internal pattern to user-readable triple format"
+  [db pattern context]
+  (let [ptype (where/pattern-type pattern)
+        pdata (where/pattern-data pattern)]
+    (case ptype
+      :class
+      (let [[s _ o] pdata]
+        {:subject (component->user-value db s context)
+         :property "@type"
+         :object (component->user-value db o context)})
+
+      (:tuple :id)
+      (let [[s p o] pdata]
+        {:subject (component->user-value db s context)
+         :property (component->user-value db p context)
+         :object (component->user-value db o context)})
+
+      ;; Other pattern types (filter, bind, etc.)
+      {:type ptype
+       :data (pr-str pdata)})))
+
+(defn- pattern-explain
+  "Generate explain information for a single pattern"
+  [db stats pattern context]
+  (let [ptype       (where/pattern-type pattern)
+        optimizable? (optimize/optimizable-pattern? pattern)
+        selectivity (when (and stats optimizable?)
```

Comment:
No need for this `when`, `optimize/calculate-selectivity` handles missing stats, and we can move the `optimizable-pattern?` check into it without any trouble.

Reply:
This was not part of the upstream change, but did simplify the version there a bit in c60b2fd8c29e9abd10d30430b7b471cabd83c9fc